app/models/form.rb

```diff
@@ -175,9 +175,10 @@ def visible_questionings
     questionings.reject{|q| q.hidden?}
   end
 
-  # returns questionings that work with sms forms and are not hidden
+  # returns hash of questionings that work with sms forms and are not hidden
   def smsable_questionings
-    questionings.reject{|q| q.hidden? || !q.question.qtype.smsable?}
+    smsable_questionings = questionings.reject{|q| q.hidden? || !q.question.qtype.smsable?}
+    smsable_questionings.index_by.with_index { |_, i| i + 1 }
```

Comment:
I think this would need to be `smsable_questionings.each_with_index.map{ |q, i| [i + 1, q] }.to_h`, which could also be written `smsable_questionings.map.with_index { |q, i| [i + 1, q] }.to_h` which might clarify a little how `index_by` replacing `map` works in the current code.